prow/cmd/branchprotector/protect_test.go

```diff
@@ -126,6 +127,14 @@ func (c fakeClient) GetBranches(org, repo string, onlyProtected bool) ([]github.
 	return b, nil
 }
 
+func (c *fakeClient) GetBranchProtection(org, repo, branch string) (*github.BranchProtection, error) {
+	ctx := org + "/" + repo + "=" + branch
+	if bp, ok := c.branchProtections[ctx]; ok {
+		return &bp, nil
+	}
+	return nil, nil
```

Comment:
The behavior for this case (no branch protection configured) doesn't match the behavior of the real client against the GitHub API since the API returns a 404 which the client considers an error.

Reply:
Thank you for the review. I will update this to match the new behaviour, i.e. empty branch protection object when branch protection is not configured.

Reply:
Fixing the client to return `nil` if the branch is not protected should fix this as well.